lib/kitchen/directions/bake_injected_exercise/bake_injected_exercise.rb

```diff
@@ -1,17 +1,71 @@
 # frozen_string_literal: true
 
 module Kitchen::Directions::BakeInjectedExercise
-  def self.v1(exercise:)
-    V1.new.bake(exercise: exercise)
+  def self.v1(exercise:, alphabetical_multiparts: false)
+    V1.new.bake(exercise: exercise, alphabetical_multiparts: alphabetical_multiparts)
   end
 
   class V1
-    def bake(exercise:)
+    def bake(exercise:, alphabetical_multiparts:)
       question_count = exercise.injected_questions.count
       exercise[:'data-question-count'] = question_count
       exercise[:'data-is-multipart'] = question_count > 1 ? 'True' : 'False'
 
       context = exercise&.exercise_context
+      stimulus = exercise&.first("div[data-type='exercise-stimulus']")
+
+      # To handle alphabetical multipart questions without interrupting numbering
+      if alphabetical_multiparts && stimulus
+        solutions_clipboard = Kitchen::Clipboard.new
+        questions_clipboard = Kitchen::Clipboard.new
+        alphabet = *('a'..'z')
+        wrapper_id = "#{exercise.injected_questions.first.id}-wrapper"
+
+        exercise.injected_questions.each_with_index do |question, index|
+          question.set(:'data-type', 'alphabetical-question-multipart')
+          question.add_class('alphabetical-multipart')
+          problem_letter = "(#{alphabet[index]})"
+          # I HAVE TO CHANGE THE DATA TYPE OF QUEStiON STEMS???
+
+          question.prepend(child:
+            <<~HTML
+              <div class="os-problem-container">
+                <a class='problem-letter' #{"href=##{question.id}-solution" if question.id.present?}>#{problem_letter}</a>
+                <span class='os-divider'> </span>
+                #{question.stimulus&.cut&.paste}
+                #{question.stem&.cut&.paste}
+              </div>
+            HTML
+          )
+
+          solution = question.solution
```

Comment:
i would put a guard before the solution logic line like `next unless solution` so that you don't have to use `&` for the rest of the chunk